From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Fri, 8 May 2015 12:02:21 +0200
Subject: Limit search in /usr/share

os.walk would walk all paths. Let's just use /usr/share/<pkg-name> for
ROS packages.
---
 src/rospkg/rospack.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rospkg/rospack.py b/src/rospkg/rospack.py
index e55acec..e21e243 100644
--- a/src/rospkg/rospack.py
+++ b/src/rospkg/rospack.py
@@ -89,7 +89,7 @@ def list_by_path(manifest_name, path, cache):
             # optimization for stacks.
             del dirs[:]
             continue #leaf     
-        elif 'rospack_nosubdirs' in files:
+        elif 'rospack_nosubdirs' in files or d.startswith('/usr/share/'):
             del dirs[:]
             continue  #leaf
         # remove hidden dirs (esp. .svn/.git)
